magma/syntax/sequential2.py

```diff
@@ -197,6 +199,96 @@ def _seq_phi(s, t, f):
     return s.ite(t, f)
 
 
+class ResetTypeSetter(ast.NodeTransformer):
+    """Set reset_type"""
+
+    def __init__(self, env, reset_type):
+        self.env = env
+        self.reset_type = reset_type
+
+    def visit_Call(self, node: ast.Call):
+        # find m.Register with no reset_type
+        if (
+            isinstance(node.func, ast.Attribute)
+            and isinstance(node.func.value, ast.Name)
+            and node.func.value.id == "m"
+            and node.func.attr == "Register"
+        ):
```

Comment:
This should work in the simple case, but there might be some other strange cases that we may want to support, for example, the user could say `from magma import Register` so there's no `m` prefix.  One thing we could do is try to serialize and evaluate the call func inside `env` and see if it's equal to m.Register, e.g. `eval(astor.to_source(node.func).rstrip()) == m.Register`

Reply:
This won't cover everything though, but I'm not sure if we can support something like:
```
def __init__(self, ...):
    Foo = m.Register
    self.x = Foo(...)
```

Reply:
Now that I think about it, maybe it's just simpler to **force** users of sequential2 to declare registers with `m.Register`, then this code should work and it would vastly simplify the task for us.